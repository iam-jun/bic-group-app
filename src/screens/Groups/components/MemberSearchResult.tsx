import {
  ActivityIndicator,
  FlatList,
  RefreshControl,
  StyleSheet,
  View,
} from 'react-native';
import React from 'react';
import {ExtendedTheme, useTheme} from '@react-navigation/native';

import Text from '~/beinComponents/Text';

import ViewSpacing from '~/beinComponents/ViewSpacing';
import MemberItem from './MemberItem';
import spacing from '~/theme/spacing';

interface MemberSearchResultProps {
  canManageMember: boolean;
  memberSearchData: {loading: boolean; canLoadMore: boolean; data: any[]};
  onLoadMore?: () => void;
  onRefresh?: () => void;
  onPressMenu: (item: any) => void;
}

const MemberSearchResult = ({
  canManageMember,
  memberSearchData,
  onLoadMore,
  onRefresh,
  onPressMenu,
}: MemberSearchResultProps) => {
  const theme: ExtendedTheme = useTheme();

  const {loading, canLoadMore, data} = memberSearchData;

  const renderItem = ({item}: {item: any}) => {
    return (
      <MemberItem
        item={item}
        canManageMember={canManageMember}
        onPressMenu={onPressMenu}
      />
    );
  };

  const renderEmptyComponent = () => {
    if (loading) return null;
    return (
      <View style={styles.textNoResults}>
        <Text.BodyM
          color={theme.colors.gray50}
          useI18n
          testID="member_search_result.no_results">
          common:text_no_results_found
        </Text.BodyM>
      </View>
    );
  };

  const renderHeaderComponent = () => {
    return (
      <View style={styles.textSearchResults}>
        <Text.BodyM useI18n>common:text_search_results</Text.BodyM>
      </View>
    );
  };

  const renderListFooter = () => {
    if (!loading && canLoadMore && data.length > 0)
      return (
        <View style={styles.listFooter}>
          <ActivityIndicator testID="member_search_result.loading_more" />
        </View>
      );

    return null;
  };

  return (
    <FlatList
      testID="flatlist"
      data={data}
      renderItem={renderItem}
      keyExtractor={(item, index) => `search_item_${item}_${index}`}
      ListHeaderComponent={renderHeaderComponent}
      ListFooterComponent={renderListFooter}
      ListEmptyComponent={renderEmptyComponent}
      showsVerticalScrollIndicator={false}
      onEndReached={onLoadMore}
      onEndReachedThreshold={0.1}
      refreshControl={
        onRefresh ? (
          <RefreshControl
            refreshing={loading}
            onRefresh={onRefresh}
            tintColor={theme.colors.gray40}
          />
        ) : undefined
      }
      ItemSeparatorComponent={() => <ViewSpacing height={4} />}
    />
  );
};

const styles = StyleSheet.create({
  textSearchResults: {
    marginHorizontal: spacing.margin.large,
    marginVertical: spacing.margin.base,
  },
  textNoResults: {
    alignItems: 'center',
    marginVertical: 100,
  },
  listFooter: {
    height: 100,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default MemberSearchResult;
