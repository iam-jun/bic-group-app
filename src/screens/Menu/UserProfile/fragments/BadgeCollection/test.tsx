import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
