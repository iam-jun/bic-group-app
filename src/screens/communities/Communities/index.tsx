import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { useState } from 'react';
import { View, StyleSheet } from 'react-native';
import Header from '~/beinComponents/Header';
import { useBaseHook } from '~/hooks';
import spacing from '~/theme/spacing';
import Tab from '~/baseComponents/Tab';
import YourCommunities from './components/YourCommunities';
import YourGroups from './components/YourGroups';
import Managed from './components/Managed';
import menuStack from '~/router/navigator/MainStack/stacks/menuStack/stack';
import { useRootNavigation } from '~/hooks/navigation';
import SearchCommunity from './components/SearchCommunity';

const HEADER_TAB = [
  { id: 'community-tab-1', text: 'communities:community_menu:your_communities' },
  { id: 'community-tab-2', text: 'communities:community_menu:your_groups' },
  { id: 'community-tab-3', text: 'communities:community_menu:managed' },
];

const Index = () => {
  const theme = useTheme();
  const { elevations } = theme;
  const styles = themeStyles(theme);
  const { t } = useBaseHook();
  const { rootNavigation } = useRootNavigation();

  const [isOpenSearchCommunity, setIsOpenSearchCommunity] = useState(false);
  const [selectedIndex, setSelectedIndex] = useState<number>(0);

  const onPressSearch = () => {
    setIsOpenSearchCommunity(true);
  };

  const onCloseSearch = () => {
    setIsOpenSearchCommunity(false);
  };

  const onPressTab = (item: any, index: number) => {
    setSelectedIndex(index);
  };

  const onDiscoverButtonPress = () => {
    rootNavigation.navigate(menuStack.discover);
  };

  const renderContent = () => {
    if (selectedIndex === 0) {
      return <YourCommunities />;
    }

    if (selectedIndex === 1) {
      return <YourGroups />;
    }

    if (selectedIndex === 2) {
      return <Managed />;
    }

    return null;
  };

  return (
    <View style={styles.containerScreen}>
      <Header
        hideBack
        removeBorderAndShadow
        title="tabs:communities"
        titleTextProps={{ useI18n: true, style: styles.textHeader }}
        icon="iconSearch"
        onPressIcon={onPressSearch}
        buttonProps={{ icon: 'CompassSolid', style: styles.buttonHeader }}
        onPressButton={onDiscoverButtonPress}
      />
      <View style={styles.containerContent}>
        <View style={[styles.containerTabView, elevations.e1]}>
          <Tab
            style={styles.tabs}
            buttonProps={{ type: 'primary', useI18n: true }}
            data={HEADER_TAB}
            type="pill"
            onPressTab={onPressTab}
            activeIndex={selectedIndex}
          />
        </View>
        {renderContent()}
      </View>
      <SearchCommunity
        isOpen={isOpenSearchCommunity}
        onClose={onCloseSearch}
        placeholder={t('communities:text_search_communities')}
      />
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    containerScreen: {
      flex: 1,
      backgroundColor: colors.white,
    },
    containerContent: {
      flex: 1,
      backgroundColor: colors.gray5,
    },
    containerTabView: {
      paddingBottom: spacing.padding.small,
      backgroundColor: colors.white,
    },
    tabs: {
      alignItems: 'center',
      paddingHorizontal: spacing.margin.large,
    },
    textHeader: {
      marginLeft: spacing.margin.small,
    },
    buttonHeader: {
      marginLeft: spacing.margin.tiny,
      marginRight: spacing.margin.small,
    },
  });
};

export default Index;
