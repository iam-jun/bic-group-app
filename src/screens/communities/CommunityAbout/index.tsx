import React from 'react';
import Header from '~/beinComponents/Header';
import ScreenWrapper from '~/beinComponents/ScreenWrapper';
import { useBaseHook } from '~/hooks';
import { useRootNavigation } from '~/hooks/navigation';
import { useKeySelector } from '~/hooks/selector';
import groupStack from '~/router/navigator/MainStack/stacks/groupStack/stack';
import groupsKeySelector from '~/storeRedux/groups/keySelector';
import AboutContent from '../CommunityDetail/components/AboutContent';
import groupJoinStatus from '~/constants/groupJoinStatus';

const CommunityAbout = () => {
  const { rootNavigation } = useRootNavigation();
  const { t } = useBaseHook();
  const infoDetail = useKeySelector(groupsKeySelector.communityDetail);
  const { name, id, joinStatus } = infoDetail;
  const isMember = joinStatus === groupJoinStatus.member;

  const onPressTotalMember = () => {
    rootNavigation.navigate(groupStack.communityMembers, { communityId: id, isMember });
  };

  return (
    <ScreenWrapper isFullView>
      <Header title={`${t('settings:title_about')} ${name}`} />
      <AboutContent profileInfo={infoDetail} onPressMember={onPressTotalMember} />
    </ScreenWrapper>
  );
};

export default CommunityAbout;
