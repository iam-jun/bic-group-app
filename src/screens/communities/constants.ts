export const MEMBER_TAB_TYPES = {
  MEMBER_LIST: 'MEMBER_LIST',
  MEMBER_REQUESTS: 'MEMBER_REQUESTS',
  INVITED_PEOPLE: 'INVITED_PEOPLE',
};
