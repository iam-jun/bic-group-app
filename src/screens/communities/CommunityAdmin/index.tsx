import { StyleSheet, View } from 'react-native';
import React from 'react';
import { ExtendedTheme, useTheme } from '@react-navigation/native';

import ScreenWrapper from '~/beinComponents/ScreenWrapper';
import Header from '~/beinComponents/Header';
import Divider from '~/beinComponents/Divider';

import { useRootNavigation } from '~/hooks/navigation';
import Text from '~/baseComponents/Text';
import MenuItem from '~/beinComponents/list/items/MenuItem';
import groupStack from '~/router/navigator/MainStack/stacks/groupStack/stack';
import spacing from '~/theme/spacing';
import useCommunitiesStore, { ICommunitiesState } from '~/store/entities/communities';
import { PermissionKey } from '~/constants/permissionScheme';
import useMyPermissionsStore from '~/store/permissions';
import articleStack from '~/router/navigator/MainStack/stacks/articleStack/stack';

const CommunityAdmin = (props: any) => {
  const { params } = props.route;
  const communityId = params?.communityId;

  const theme: ExtendedTheme = useTheme();
  const styles = createStyles();
  const { rootNavigation } = useRootNavigation();
  const community = useCommunitiesStore((state: ICommunitiesState) => state.data[communityId]);
  const { name = '', groupId } = community || {};

  const { shouldHavePermission } = useMyPermissionsStore((state) => state.actions);
  const canEditProfileInfo = shouldHavePermission(
    groupId,
    [
      PermissionKey.EDIT_INFO,
      PermissionKey.EDIT_PRIVACY,
    ],
  );

  const onPressGeneralInfo = () => {
    rootNavigation.navigate(
      groupStack.generalInfo, {
        id: communityId,
        type: 'community',
      },
    );
  };

  const onPressScheduleContent = () => {
    rootNavigation.navigate(
      articleStack.articleScheduleContent, { groupId },
    );
  };

  const renderModerating = () => null;
  // return (
  //   <>
  //     <Text.BodyM
  //       style={styles.headerTitle}
  //       color={theme.colors.neutral80}
  //       variant="bodyM"
  //       useI18n
  //     >
  //       settings:title_community_moderating
  //     </Text.BodyM>
  //     <MenuItem
  //       testID="community_admin.pending_posts"
  //       title="settings:title_pending_posts"
  //       icon="FileExclamation"
  //       iconProps={{
  //         icon: 'FileExclamation',
  //         tintColor: theme.colors.purple50,
  //       }}
  //       notificationsBadgeNumber={999}
  //       notificationsBadgeProps={{ maxNumber: 99, variant: 'alert' }}
  //       rightSubIcon="AngleRightSolid"
  //       onPress={displayNewFeature}
  //     />
  //   </>
  // );

  const renderSettings = () => (
    <>
      <Text.BodyM
        style={styles.headerTitle}
        color={theme.colors.neutral80}
        variant="bodyM"
        useI18n
      >
        settings:title_community_settings
      </Text.BodyM>
      {!!canEditProfileInfo && (
        <MenuItem
          testID="community_admin.profile_info"
          title="settings:title_profile_info"
          icon="Gear"
          iconProps={{ icon: 'Gear', tintColor: theme.colors.purple50 }}
          rightSubIcon="AngleRightSolid"
          onPress={onPressGeneralInfo}
        />
      )}
      <MenuItem
        testID="community_admin.schedule_content"
        title="settings:title_schedule_content"
        icon="BallotCheck"
        iconProps={{ icon: 'BallotCheck', tintColor: theme.colors.purple50 }}
        rightSubIcon="AngleRightSolid"
        onPress={onPressScheduleContent}
      />
    </>
  );

  return (
    <ScreenWrapper testID="CommunityAdmin" isFullView>
      <Header
        title={name}
        titleTextProps={{ color: theme.colors.neutral80 }}
      />
      <Divider size={4} />
      <View style={styles.container}>
        {renderModerating()}
        {renderSettings()}
      </View>
    </ScreenWrapper>
  );
};

const createStyles = () => StyleSheet.create({
  container: {
    paddingTop: spacing.padding.large,
  },
  headerTitle: {
    marginHorizontal: spacing.margin.large,
    marginVertical: spacing.margin.base,
  },
});

export default CommunityAdmin;
