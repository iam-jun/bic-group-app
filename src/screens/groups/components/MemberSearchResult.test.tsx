import React from 'react';

import { renderWithRedux } from '~/test/testUtils';
import MemberSearchResult from './MemberSearchResult';
import { adminDetail, memberDetail } from '~/test/mock_data/communities';

describe('MemberSearchResult component', () => {
  const onPressMenu = jest.fn();
  const onLoadMore = jest.fn();

  it('should NOT render empty screen correctly when loading', () => {
    const wrapper = renderWithRedux(
      <MemberSearchResult
        canManageMember
        memberSearchData={{ loading: true, canLoadMore: true, data: [] }}
        isAdminRole
        onPressMenu={onPressMenu}
        onLoadMore={onLoadMore}
      />,
    );
    const emptyText = wrapper.queryByTestId('member_search_result.no_results');
    expect(emptyText).toBeNull();
  });

  it('should NOT render empty screen correctly when having data and NOT loading', () => {
    const wrapper = renderWithRedux(
      <MemberSearchResult
        canManageMember
        memberSearchData={{
          loading: false,
          canLoadMore: true,
          data: [adminDetail, memberDetail],
        }}
        isAdminRole
        onPressMenu={onPressMenu}
        onLoadMore={onLoadMore}
      />,
    );
    const emptyText = wrapper.queryByTestId('member_search_result.no_results');
    expect(emptyText).toBeNull();
  });

  it('should render empty screen correctly when having NO data and NO loading', () => {
    const wrapper = renderWithRedux(
      <MemberSearchResult
        canManageMember
        memberSearchData={{
          loading: false,
          canLoadMore: true,
          data: [],
        }}
        isAdminRole
        onPressMenu={onPressMenu}
        onLoadMore={onLoadMore}
      />,
    );
    const emptyText = wrapper.getByTestId('no_search_results');
    expect(emptyText).toBeDefined();
  });

  it('should render loading more indicator correctly', () => {
    const wrapper = renderWithRedux(
      <MemberSearchResult
        canManageMember
        memberSearchData={{
          loading: false,
          canLoadMore: true,
          data: [adminDetail, memberDetail],
        }}
        isAdminRole
        onPressMenu={onPressMenu}
        onLoadMore={onLoadMore}
      />,
    );
    const loadingIndicator = wrapper.getByTestId(
      'member_search_result.loading_more',
    );
    expect(loadingIndicator).toBeDefined();
  });

  it('should NOT render loading more indicator correctly', () => {
    const wrapper = renderWithRedux(
      <MemberSearchResult
        canManageMember
        memberSearchData={{
          loading: true,
          canLoadMore: true,
          data: [],
        }}
        isAdminRole
        onPressMenu={onPressMenu}
        onLoadMore={onLoadMore}
      />,
    );
    const loadingIndicator = wrapper.queryByTestId(
      'member_search_result.loading_more',
    );
    expect(loadingIndicator).toBeNull();
  });

  it('should render data correctly', () => {
    const wrapper = renderWithRedux(
      <MemberSearchResult
        canManageMember
        memberSearchData={{
          loading: false,
          canLoadMore: false,
          data: [adminDetail, memberDetail],
        }}
        isAdminRole
        onPressMenu={onPressMenu}
        onLoadMore={onLoadMore}
      />,
    );
    const flatlist = wrapper.getByTestId('flatlist');
    expect(flatlist).toBeDefined();
    const items = wrapper.getAllByTestId('member_item');
    expect(items.length).toBe(2);
  });
});
