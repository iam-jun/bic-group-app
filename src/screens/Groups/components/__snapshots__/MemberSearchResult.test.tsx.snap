// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberSearchResult component should render data correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={
    Array [
      Object {
        "avatar": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/e41d70bc-0bfd-4c56-a461-c2dfb10434c1.jpg",
        "chat_user_id": "tg6kmc6cmybgwpcif9znngyyebe",
        "fullname": "Test Name",
        "id": 1,
        "roles": Object {
          "name": "COMMUNITY_ADMIN",
        },
        "username": "test",
      },
      Object {
        "avatar": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/e41d70bc-0bfd-4c56-a461-c2dfb10434c1.jpg",
        "chat_user_id": "tg6kmc6cmybgwpcif9znngyyebe",
        "fullname": "Test Name",
        "id": 1,
        "roles": Object {
          "name": "MEMBER",
        },
        "username": "test",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={0.1}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  testID="flatlist"
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "marginHorizontal": 16,
            "marginVertical": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": undefined,
                "fontFamily": "BeVietnamPro-Light",
                "fontSize": 15,
              },
              Object {},
              undefined,
            ]
          }
        >
          Search Results
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": undefined,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 4,
            }
          }
          testID="member_item"
        >
          <View
            style={
              Object {
                "borderRadius": 4,
                "height": 40,
                "marginRight": 12,
                "width": 40,
              }
            }
            testID="avatar"
          >
            <View
              style={
                Object {
                  "borderRadius": 20,
                  "height": 40,
                  "width": 40,
                }
              }
              testID="avatar_container"
            >
              <View
                onLayout={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "position": "relative",
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#f9f9f9",
                        "borderRadius": 20,
                        "height": 40,
                        "justifyContent": "center",
                        "width": 40,
                      }
                    }
                    testID="Image__placeholder"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 20,
                        "height": 40,
                        "width": 40,
                      },
                    ]
                  }
                >
                  <FastImageView
                    onFastImageError={[Function]}
                    onFastImageLoadEnd={[Function]}
                    resizeMode="cover"
                    source={
                      Object {
                        "uri": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/e41d70bc-0bfd-4c56-a461-c2dfb10434c1.jpg",
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                    testID="avatar.image"
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontFamily": "BeVietnamPro-Light",
                    "fontSize": 15,
                  },
                  Object {},
                  undefined,
                ]
              }
            >
              Test Name
              <Text
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontFamily": "BeVietnamPro-Light",
                      "fontSize": 13,
                    },
                    Object {},
                    undefined,
                  ]
                }
              >
                 @test
              </Text>
            </Text>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="member_item.icon_chat.button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "alignItems": "center",
                    "height": 36,
                    "justifyContent": "center",
                    "marginLeft": 4,
                    "padding": 8,
                    "width": 36,
                  },
                  Object {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View />
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="member_item.icon_option.button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "marginLeft": 8,
                  },
                  Object {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 4,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "width": undefined,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": undefined,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 4,
            }
          }
          testID="member_item"
        >
          <View
            style={
              Object {
                "borderRadius": 4,
                "height": 40,
                "marginRight": 12,
                "width": 40,
              }
            }
            testID="avatar"
          >
            <View
              style={
                Object {
                  "borderRadius": 20,
                  "height": 40,
                  "width": 40,
                }
              }
              testID="avatar_container"
            >
              <View
                onLayout={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "position": "relative",
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#f9f9f9",
                        "borderRadius": 20,
                        "height": 40,
                        "justifyContent": "center",
                        "width": 40,
                      }
                    }
                    testID="Image__placeholder"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 20,
                        "height": 40,
                        "width": 40,
                      },
                    ]
                  }
                >
                  <FastImageView
                    onFastImageError={[Function]}
                    onFastImageLoadEnd={[Function]}
                    resizeMode="cover"
                    source={
                      Object {
                        "uri": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/e41d70bc-0bfd-4c56-a461-c2dfb10434c1.jpg",
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                    testID="avatar.image"
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontFamily": "BeVietnamPro-Light",
                    "fontSize": 15,
                  },
                  Object {},
                  undefined,
                ]
              }
            >
              Test Name
              <Text
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontFamily": "BeVietnamPro-Light",
                      "fontSize": 13,
                    },
                    Object {},
                    undefined,
                  ]
                }
              >
                 @test
              </Text>
            </Text>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="member_item.icon_chat.button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "alignItems": "center",
                    "height": 36,
                    "justifyContent": "center",
                    "marginLeft": 4,
                    "padding": 8,
                    "width": 36,
                  },
                  Object {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View />
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="member_item.icon_option.button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "marginLeft": 8,
                  },
                  Object {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    />
  </View>
</RCTScrollView>
`;
